trackmania.js

```diff
@@ -537,50 +584,129 @@ export class TrackmaniaFacade {
                 description: `Leaderboard positions ${Number(offset)+(Number(length)*i)} - ${(Number(offset)+Number(length))+(Number(length)*i)}`,
             });
         }
-        
-        let records = [];
-        let s_accounts = [];
-        if (lb_info.accountIds.length > 0) {
-            const accounts = await this.oauthService.fetchAccountNames(lb_info.accountIds);
-            let field = {
-                name: `${lb_info.positions[0]} - ${lb_info.positions[lb_info.positions.length - 1]}`,
-                value: '```',
-                inline: true,
-            };
-            lb_info.accountIds.forEach((accountId, i) => {
-                field.value += `${lb_info.positions[i].toString().padStart(5)}: ${lb_info.times[i]} - ${accounts[accountId]}\n`;
-                s_accounts.push({
-                    label: `${accounts[accountId]}`,
-                    value: `${accountId}`,
-                });
-            });
-            field.value += '```';
-            records.push(field);
-        }
-        else {
-            records.push({
-                name: `no records available`,
-                value: `come back later :(`,
-                inline: true,
-            });
-        }
 
-        const { groupUid, timestamp } = toBase64(track_info.groupUid, track_info.endTimestamp);
+        const recordsJSON = {
+            name: `${pages[0].label} : ${lb_info[0].position} - ${lb_info[lb_info.length-1].position}`,
+            records: lb_info,
+        };
+
+        const records = await this.generateLeaderboardField(recordsJSON);
 
-        const leaderboard_info = {
-            groupUid: groupUid,
-            timestamp: timestamp,
-            mapUid: track_info.mapUid,
-            length: Number(length),
-            offset: Number(offset),
+        let buttons = [{
```

Comment:
Why does all of this button construction need to live in `updateLeaderboardInfo`? Why can't we leave it in `embedLeaderboardInfo`?